src/libbson/doc/bson_utf8_escape_for_json.rst

```diff
@@ -32,3 +32,7 @@ Returns
 -------
 
 A newly allocated string that should be freed with :symbol:`bson_free()`.
+
+.. warning::
+  If ``utf8`` represents invalid UTF-8 data :symbol:`bson_utf8_escape_for_json` may return NULL or have undefined behavior.
+  UTF-8 may be validated with :symbol:`bson_utf8_validate`.
```

Comment:
```suggestion
A newly allocated string that should be freed with :symbol:`bson_free()` when ``utf8`` is a valid UTF-8 string, or ``NULL`` if the (possibly invalid UTF-8) string could not be escaped.
```

We should not expose undefined behavior to users when it is (easily) avoidable. Given this function is already capable of handling invalid inputs, recommend documenting the `NULL` result condition as a wider and simpler "could not be escaped" instead.